src/oc_erchef/apps/chef_index/src/chef_index_query.erl

```diff
@@ -106,7 +106,7 @@ decode({nonneg_int, Key}, Val, Default) ->
                 {Default, default};
             Value ->
                 try
-                    {list_to_integer(http_uri:decode(Value)), Value}
```

Comment:
Any reason for removing the http_uri:decode completely ? Does it effect any functionality ?

Reply:
The http_uri:decode function was either deprecated or removed.  I fixed the issue, and then made further changes that @marcparadise recommended in this PR.

It builds and passes tests.

Reply:
While it is safe/correct to skip the decode in casese where we're just doing a `list_to_integer` anyway, I did not see our usage of http_uri:decode in the context of `transform_query` where I think we still need that functionality for complex/multipart queries.  

Maybe our best answer for this case is to pull in the original http_uri implementation of `decode`:
 
https://github.com/erlang/otp/blob/maint-21/lib/inets/src/http_lib/http_uri.erl#L143-L161

Reply:
@mp Done.  Feel free to have a look.

Reply:
Since we have pulled in the original implementation, should we leave the calls to decode and unencode as is?

Reply:
@PrajaktaPurohit Marc thought it would be safe, plus it would simplify things, if we eliminated some of the calls to decode.  But others are left alone.